### PR TITLE
fix: predicted bill shows number over limit

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

If a customer's predicted usage was greater than that of their limit, the UI would say that their predicted bill was higher than their limit. Eg:

image

This is confusing because their actual predicted bill is just the limit.

## Changes

If the billing limit is less than the current predicted amount for the product, then say that the limit is the predicted bill. I've also included the actual predicted amount if they were unlimited in the tooltip when you hover over the predicted bill.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

👀

### CODE REVIEW

**Review by charlottemcginn — COMMENTED** [review]
I wonder if there's a way for us to show the projected amount still without hovering over the tooltip to emphasize "this is how many events we are expecting you to drop." Ideally we want customers to increase their usage limits and I do think visually emphasizing the data they will miss out on is important.

One idea -- maybe we could instead have the line between "Billing limit" and "Projected" be a dashed dark grey line, and add a caption below that line that says "projected dropped events"

**Comment by raquelmsmith** [comment]
> I wonder if there's a way for us to show the projected amount still without hovering over the tooltip to emphasize "this is how many events we are expecting you to drop." Ideally we want customers to increase their usage limits and I do think visually emphasizing the data they will miss out on is important.
>
> One idea -- maybe we could instead have the line between "Billing limit" and "Projected" be a dashed dark grey line, and add a caption below that line that says "projected dropped events"

Was having a hard time visualizing what you meant here, but I get where you are coming from.

The new UI is a bit different, could do something like this? 
image

**Comment by charlottemcginn** [comment]
236324147-de2aac02-fab1-4e7f-a041-1bff8c72c88a

(a little janky but it works) this is more what I was thinking of, I just feel like it better communicates *what you're missing out on*

edit: I realized I can't read and you were saying *predicted* and not *projected* and thought we were getting rid of the *projected* upper limit when a Billing Limit was set. Sorry about that!

File: frontend/src/scenes/billing/BillingProduct.tsx [hunk 1]
```diff
@@ -200,13 +203,21 @@ const BillingProduct = ({ product }: { product: BillingProductV2Type }): JSX.Ele
...
                                         Predicted bill
                                     </LemonLabel>
                                     <div className="font-bold text-muted text-2xl">
-                                        ${product.projected_amount_usd || '0.00'}
+                                        $
+                                        {
+                                            // if the billing limit for the product is less than the projected amount, show the limit instead
+                                            predictUsageWillBeLimited
+                                                ? billing?.custom_limits_usd?.[product.type]
+                                                : product.projected_amount_usd || '0.00'
```
Comment (charlottemcginn) [inline frontend/src/scenes/billing/BillingProduct.tsx:219]: do we also need to check against how much they've already used, in the case that they set their custom limit to be less than what they've used for the month?

**Comment by raquelmsmith** [comment]
ohhhh I see what you mean. Yeah the current gauge will stay the same, but I do like the idea of showing how much will get lost. I think changing how the colors are used will help with that a lot, it's something I've been meaning to do.

File: frontend/src/scenes/billing/BillingProduct.tsx (hunk 1 above)
Comment (raquelmsmith) [reply frontend/src/scenes/billing/BillingProduct.tsx:219]: Ah, you're right. 😑 will update.

**Pull request closed without merging**